ada_2_fhir-r4/zibs2020/payload/0.0/nl-core-MedicalDevice.xsl

```diff
@@ -0,0 +1,369 @@
+                        <xsl:when test="(not($startDate) or nf:isPast($startDate)) and nf:isPast($endDate)">
+                            <xsl:attribute name="value" select="'completed'"/>
+                        </xsl:when>
+                        
+                        <!-- If no status can be derived from the startDate and endDate, the MedicalDevice is assumed to be active. 
+                            A status code must be provided and no unknown code exists in the required ValueSet.-->
+                        <xsl:otherwise>
+                            <xsl:attribute name="value" select="'active'"/>
+                        </xsl:otherwise>
+                    </xsl:choose>
+                </status>
+                      
+                <xsl:call-template name="makeReference">
+                    <xsl:with-param name="in" select="$subject"/>
+                    <xsl:with-param name="wrapIn" select="'subject'"/>
+                </xsl:call-template>
+                
+                <xsl:if test="$startDate or $endDate">
+                    <timingPeriod>
+                        <xsl:for-each select="begin_datum">
+                            <start>
+                                <xsl:call-template name="date-to-datetime">
+                                    <xsl:with-param name="in" select="."/>
+                                </xsl:call-template>    
+                            </start>
+                        </xsl:for-each>
+                        <xsl:for-each select="eind_datum">
+                            <end>
+                                <xsl:call-template name="date-to-datetime">
+                                    <xsl:with-param name="in" select="."/>
+                                </xsl:call-template>    
+                            </end>
+                        </xsl:for-each>
+                    </timingPeriod>
+                </xsl:if>
+
+                <xsl:for-each select="product">
+                    <device>
+                        <xsl:call-template name="makeReference">
+                            <xsl:with-param name="profile" select="concat($profile,'.Product')"/>
+                        </xsl:call-template>
+                    </device>
+                </xsl:for-each>
+                    
+                <xsl:for-each select="indicatie">
+                    <xsl:call-template name="makeReference">
+                        <xsl:with-param name="in" select="probleem"/>
+                        <xsl:with-param name="wrapIn" select="'reasonReference'"/>
+                    </xsl:call-template>
+                </xsl:for-each>
+                
+                <!--The element reasonReference is present to support Observations that refer to a MedicalDevice, such as HearingFunction and VisualFunction. -->
+                <xsl:if test="$reasonReference">
+                    <xsl:for-each select="../..">
```

Comment:
I wonder if this works in general (for HearingFunction and VisualFunction it works) or if some ada instances that refer to a MedicalDevice have another structure. Should we implement this as a variable instead, or check each time a mapping is made that includes a MedicalDevice whether this logic is still appropriate?

Reply:
Fixed